funderfinder/sources/github_sponsors.py

```diff
@@ -62,16 +73,107 @@ def parse_gh_org_funding_json(gh_org_funding_json: Any) -> dict:
     return funding_dict
 
 
+def get_gh_top_contributors_json(gh_url: str, num_top_contribs: int = 3) -> list:
+    """
+    Retrieves JSON of top contributors to a GitHub repo and whether these
+    contributors have GitHub sponsors. Needs to use GitHub REST API because
+    GraphQL API doesn't return contributors (as of Dec. 2022).
+    :param gh_url: GitHub repository URL
+    :param num_top_contribs: number of contributors to check
+    :return: list
+    """
+    org_and_owner = get_org_and_owner_from_github_url(gh_url)
+    response = requests.get(
+        "https://api.github.com/repos/"
+        + org_and_owner
+        + "/contributors?page=1"
+        + "&per_page="
+        + str(num_top_contribs),
+        # convert username and token to strings per requests's specifications
+        auth=(str(os.environ.get("GITHUB_USERNAME")), os.environ.get("GITHUB_TOKEN")),
+    )
+
+    if response.ok:
+        contributors = json.loads(response.text or response.content)
+
+    top_contribs = []
+    for contributor in contributors:
+        top_contribs.append(contributor["login"])
+
+    return top_contribs
+
+
+def get_gh_user_gh_sponsors(user: str) -> Any:
+    """
+    Retrieves GitHub sponsors JSON for a GitHub organization. See:
+    :param user: identifier a GitHub user
+    :return: JSON
+    """
+    query = """
+        query ($user: String!) {
+            user(login: $user) {
+                sponsors(first: 100) {
```

Comment:
We'll probably eventually want to traverse all the pages of results, right? (if I'm not misreading that we're only grabbing the top 100 for now)

Reply:
Good point. It should and is not currently. Here is some starter code: https://til.simonwillison.net/github/graphql-pagination-python

My lazy judgment is that there will rarely be more than 100 sponsors for a given user. Why? Because `ljharb` is arguably the most prolific npm maintainer (he is responsible for packages whose download counts account for 10 percent of npm downloads) and not even he has 100 sponsors.

I'll open an issue, but since this is a little tricky, I would vote for moving on to other features before cleaning this up. wdyt?

Reply:
Sounds good